File: algorithms/code/leetcode/lc190_reverse_bits/lc190_reverse_bits.py
# https://leetcode.com/problems/reverse-bits/

import unittest


class Solution:
    def reverseBits(self, n: int) -> int:
        r = 0
        for _ in range(0, 32):
            r <<= 1
            r |= n & 1
            n >>= 1
        return r


class TestCode(unittest.TestCase):
    def test_43261596(self) -> None:
        assert Solution().reverseBits(43261596) == 964176192

File: algorithms/code/leetcode/lc190_reverse_bits/lc190_reverse_bits.rs
// https://leetcode.com/problems/reverse-bits/
// #easy

pub struct Solution;

impl Solution {
    pub fn reverse_bits(mut n: i32) -> i32 {
        let mut r = 0;
        for _ in 0..32 {
            r <<= 1;
            r |= n & 1;
            n >>= 1;
        }
        r
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_43261596() {
        assert_eq!(Solution::reverse_bits(43261596), 964176192);
    }
}
